src/Altinn.App.Core/Features/Validation/GenericFormDataValidator.cs

```diff
@@ -58,13 +58,15 @@ public bool HasRelevantChanges(object current, object previous)
     /// <param name="description">Optional description if you want to provide a user friendly message that don't rely on the translation system</param>
     /// <param name="code">optional short code for the type of issue</param>
     /// <param name="customTextParams">List of parameters to replace after looking up the translation. Zero indexed {0}</param>
+    /// <param name="customTextParameters">Dictionary of parameters to replace after looking up the translation.</param>
     protected void CreateValidationIssue<T>(
         Expression<Func<TModel, T>> selector,
         string textKey,
         ValidationIssueSeverity severity = ValidationIssueSeverity.Error,
         string? description = null,
         string? code = null,
-        List<string>? customTextParams = null
+        List<string>? customTextParams = null,
+        Dictionary<string, string>? customTextParameters = null
```

Comment:
Can you search for the usage of this parameter? If only a few, I think it is OK to support the List version here. It just adds confusion.

Reply:
Just change the type as you did in `ITranslationService`. The previous code didn't work anyway, and those getting a build failure will need a notice that "Now it works".

Reply:
Another reason to not attempt compatibility